src/cfn-resource-mappings.ts

```diff
@@ -115,14 +113,10 @@ export function mapToCfnResource(
         }
 
         default: {
-            // When creating a generic `CfnResource` we don't have any information on the
-            // attributes attached to the resource. We need to populate them by looking up the
-            // `output` in the metadata
-            const metadata = new Metadata(PulumiProvider.AWS_NATIVE);
-            const resource = metadata.findResource(typeName);
-            const attributes = Object.keys(resource.outputs);
-
-            return new CfnResource(logicalId, typeName, props, attributes, options);
+            const mName = moduleName(typeName).toLowerCase();
+            const pType = pulumiTypeName(typeName);
+            const awsModule = aws as any;
+            return new awsModule[mName][pType](logicalId, props, options);
```

Comment:
Neat!
This only works for resources that are nested in a module. Which is fine as there's only the `ExtensionResource` on the root level, which we do not use, and all CFN resources are nested in a module!